internal/subscription/database/subscription_queries.go

```diff
@@ -1,26 +1,97 @@
+		createSubscription: fmt.Sprintf(`INSERT INTO subscriptions (
+				course_id, matrix_id, user_uuid, role, expires_at
+			) SELECT
+				courses.id, matrices.id, $3, $4, $5
+			FROM courses, matrices
+			WHERE
+				courses.uuid = $1 AND courses.deleted_at IS NULL
+				AND matrices.uuid = $2 AND matrices.deleted_at IS NULL
+			RETURNING %s`, returningColumns),
+		createSubscriptionWithoutMatrix: fmt.Sprintf(`INSERT INTO subscriptions (
+				course_id, user_uuid, role, expires_at
+			) SELECT
+				courses.id, $2, $3, $4
+			FROM courses
+			WHERE
+				courses.uuid = $1 AND courses.deleted_at IS NULL
+			RETURNING %s`, returningColumns),
+		deleteSubscription: fmt.Sprintf(`UPDATE subscriptions
+			SET deleted_at = NOW(), reason = $2
+			WHERE uuid = $1 AND deleted_at IS NULL
+			RETURNING %s, reason, deleted_at`, returningColumns),
+		getSubscription: fmt.Sprintf(`SELECT
+				courses.uuid AS course_uuid,
+				matrices.uuid AS matrix_uuid,
+				%s
+			FROM subscriptions
+				LEFT JOIN courses ON subscriptions.course_id = courses.id
+				LEFT JOIN matrices ON subscriptions.matrix_id = matrices.id
+			WHERE
+				subscriptions.uuid = $1 AND subscriptions.deleted_at IS NULL
+				AND courses.deleted_at IS NULL
+				AND matrices.deleted_at IS NULL`, returningColumns),
+		listSubscription: fmt.Sprintf(`SELECT
+				courses.uuid AS course_uuid,
+				matrices.uuid AS matrix_uuid,
+				%s
+			FROM subscriptions
+				LEFT JOIN courses ON subscriptions.course_id = courses.id
+				LEFT JOIN matrices ON subscriptions.matrix_id = matrices.id
+			WHERE
+				subscriptions.deleted_at IS NULL
+				AND courses.deleted_at IS NULL
+				AND matrices.deleted_at IS NULL`, returningColumns),
+		updateSubscription: fmt.Sprintf(`UPDATE subscriptions
+			SET role = $2, expires_at = $3, updated_at = NOW()
+			WHERE uuid = $1 AND deleted_at IS NULL
+			RETURNING %s`, returningColumns),
+		courseSubscriptions: fmt.Sprintf(`SELECT
+				%s,
+				matrices.uuid AS "matrices.uuid",
+				matrices.code AS "matrices.code",
+				matrices.name AS "matrices.name"
+			FROM subscriptions
+				LEFT JOIN courses ON subscriptions.course_id = courses.id
+				LEFT JOIN matrices ON subscriptions.matrix_id = matrices.id
+			WHERE courses.uuid = $1 AND courses.deleted_at IS NULL
+				AND subscriptions.deleted_at IS NULL
+				AND matrices.deleted_at IS NULL`, returningColumns),
+		userSubscriptions: fmt.Sprintf(`SELECT
+				%s,
+				courses.uuid AS "courses.uuid",
+				courses.code AS "courses.code",
+				courses.name AS "courses.name",
+				matrices.uuid AS "matrices.uuid",
+				matrices.code AS "matrices.code",
+				matrices.name AS "matrices.name"
+			FROM subscriptions
+				LEFT JOIN courses ON subscriptions.course_id = courses.id
+				LEFT JOIN matrices ON subscriptions.matrix_id = matrices.id
+			WHERE subscriptions.user_uuid = $1 AND subscriptions.deleted_at IS NULL
+				AND matrices.deleted_at IS NULL
+				AND courses.deleted_at IS NULL`, returningColumns),
```

Comment:
@rluders any suggestion?

Reply:
@marianysilva IDK... doesn't seems to be a problem for me in this case, it is just because the length of the queries, but is is very clear and easy to understand.